src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_frame.py

```diff
@@ -128,7 +128,9 @@ def should_stop_on_exception(self, frame, event, arg):
                             return False, frame
 
                     if exception_breakpoint.ignore_libraries:
-                        if not main_debugger.is_exception_trace_in_project_scope(trace):
+                        if not main_debugger.is_exception_trace_in_project_scope(trace) and \
+                            main_debugger.plugin is not None and \
```

Comment:
I think this would fail in the following case:
- add django exception breakpoint
- mark to ignore exceptions in libraries
- non-django exception is raised
- exception would be shown to the user

The current problem is the following: if there's an exception breakpoint it always takes priority over plugin breakpoints (i.e.: `if exception_breakpoint is not None...`). 

This is backwards, I think that the proper fix is checking if there are exception breakpoints and if there are, ask the plugins to handle it (i.e.: the `main_debugger.plugin.exception_break` currently in the `else` block) and if it's not handled by the plugins then go on to the current code.

So, we should have a test-case for this case (not sure if there is already) and the fix is something as:

```
# Plugins should have priority when handling exceptions so that an exception is not
# ignored if a plugin would handle it.
should_stop = False
if main_debugger.plugin is not None and main_debugger.plugin.has_exception_breaks():
    result = main_debugger.plugin.exception_break(main_debugger, self, frame, self._args, arg)
    if result:
        should_stop, frame = result

if not should_stop:
    exception_breakpoint = main_debugger.get_exception_breakpoint(
        exception, main_debugger.break_on_caught_exceptions)
    if exception_breakpoint is not None:
        ... current code to handle exception_breakpoint ...
```

Reply:
I tried this, it does not seem to work right. I see another problem. Template exceptions in django are not reporting in the django exception break context. Also the first frame is app.py in the test case. So there is something broken there.

Also, since it is not the first frame, just_reported is false. so django excpetion_break always returns None.

Reply:
So regular django-exception breakpoints aren't working properly? If something is broken in `pydevd_plugins.django_debug.exception_break` it should probably be fixed first then (do you want me to take a look at it?).

Reply:
Yes, please take a look at this. I will close this PR. I will reopen after the django part is fixed.

Reply:
Ok, created https://github.com/Microsoft/ptvsd/issues/1181 for that.